File: assignments/elixir/leap/leap_test.exs
if System.get_env("EXERCISM_TEST_EXAMPLES") do
  Code.load_file("example.exs")
else
  Code.load_file("year.exs")
end

ExUnit.start

defmodule LeapTest do
  use ExUnit.Case, async: true
  doctest Year

  test "vanilla leap year" do
    assert Year.leap_year?(1996)
  end

  test "any old year" do
    # assert ! Year.leap_year?(1997)
  end

  test "century" do
    # assert ! Year.leap_year?(1900)
  end

  test "exceptional century" do
    # assert Year.leap_year?(2400)
  end
end

File: assignments/go/leap/leap_test.go
package leap

import (
	"testing"
)

var testCases = []struct {
	year        int
	expected    bool
	description string
}{
	{1996, true, "a vanilla leap year"},
	{1997, false, "a normal year"},
	{1900, false, "a century"},
	{2400, true, "an exceptional century"},
}

func TestLeapYears(t *testing.T) {
	for _, test := range testCases {
		observed := IsLeapYear(test.year)
		if observed != test.expected {
			t.Fatalf("%v is %s", test.year, test.description)
		}
	}
}
